src/helpers/ens.ts

```diff
@@ -0,0 +1,518 @@
+    ENSReverseRegistrarABI,
+    wallet || web3Provider
+  );
+};
+
+const getENSBaseRegistrarImplementationContract = (wallet?: Wallet) => {
+  return new Contract(
+    ensBaseRegistrarImplementationAddress,
+    ENSBaseRegistrarImplementationABI,
+    wallet || web3Provider
+  );
+};
+
+const getResolver = async (name: string): Promise<string> =>
+  getENSRegistryContract().resolver(name);
+
+const getAvailable = async (name: string): Promise<boolean> =>
+  getENSRegistrarControllerContract().available(name);
+
+const getNameExpires = async (name: string): Promise<string> =>
+  getENSBaseRegistrarImplementationContract().nameExpires(
+    addHexPrefix(sha3(name))
+  );
+
+const getRentPrice = async (name: string, duration: number): Promise<any> =>
+  getENSRegistrarControllerContract().rentPrice(name, duration);
+
+const setupMulticallRecords = (
+  name: string,
+  records: ENSRegistrationRecords,
+  resolverInstance: Contract
+): string[] => {
+  const resolver = resolverInstance.interface;
+  const namehash = hash(name);
+
+  const data = [];
+
+  // ens associated address
+  const ensAssociatedRecord = records.ensAssociatedAddress;
+  if (
+    Boolean(ensAssociatedRecord) &&
+    typeof ensAssociatedRecord === 'string' &&
+    parseInt(ensAssociatedRecord, 16) !== 0
+  ) {
+    data.push(
+      resolver.encodeFunctionData('setAddr(bytes32,address)', [
+        namehash,
+        ensAssociatedRecord,
+      ])
+    );
+  }
+  // content hash address
+  const contentHashAssociatedRecord = records.contentHash;
+  if (
+    Boolean(contentHashAssociatedRecord) &&
+    typeof contentHashAssociatedRecord === 'string' &&
+    parseInt(contentHashAssociatedRecord, 16) !== 0
+  ) {
+    data.push(
+      resolver.encodeFunctionData('setContenthash', [
+        namehash,
+        contentHashAssociatedRecord,
+      ])
+    );
+  }
+  // coin addresses
+  const coinAddressesAssociatedRecord = records.coinAddress;
+  if (coinAddressesAssociatedRecord) {
+    data.push(
+      coinAddressesAssociatedRecord.map(coinRecord => {
+        const { decoder, coinType } = formatsByName[coinRecord.key];
+        let addressAsBytes;
+        if (!coinRecord.address || coinRecord.address === '') {
+          addressAsBytes = Buffer.from('');
+        } else {
+          addressAsBytes = decoder(coinRecord.address);
+        }
+        return resolver.encodeFunctionData('setAddr(bytes32,uint256,bytes)', [
+          namehash,
+          coinType,
+          addressAsBytes,
+        ]);
+      })
+    );
+  }
+  // text addresses
+  const textAssociatedRecord = records.text;
+  if (textAssociatedRecord) {
+    data.push(
+      textAssociatedRecord.map(textRecord => {
+        return resolver.encodeFunctionData('setText', [
+          namehash,
+          textRecord.key,
+          textRecord.value,
+        ]);
+      })
+    );
+  }
+  // flatten textrecords and addresses and remove undefined
+  return data.flat().filter(Boolean);
```

Comment:
multicall tx, for setting multiple records, receives an array of operations to execute